great_expectations/data_context/store/datasource_store.py

```diff
@@ -62,20 +64,40 @@ def remove_key(self, key: Union[DataContextVariableKey, GXCloudIdentifier]) -> N
         """
         return self._store_backend.remove_key(key.to_tuple())
 
-    def serialize(self, value: DatasourceConfig) -> Union[str, dict, DatasourceConfig]:
+    def serialize(
+        self, value: DatasourceConfig | FluentDatasource
+    ) -> Union[str, dict, DatasourceConfig]:
         """
         See parent 'Store.serialize()' for more information
         """
+        if isinstance(value, FluentDatasource):
+            return value._json_dict()
         return self._serializer.serialize(value)
 
-    def deserialize(self, value: Union[dict, DatasourceConfig]) -> DatasourceConfig:
+    @overload
+    def deserialize(self, value: DatasourceConfig) -> DatasourceConfig:
+        ...
+
+    @overload
+    def deserialize(self, value: FluentDatasource) -> FluentDatasource:
+        ...
+
+    def deserialize(
+        self, value: dict | DatasourceConfig | FluentDatasource
+    ) -> DatasourceConfig | FluentDatasource:
         """
         See parent 'Store.deserialize()' for more information
         """
         # When using the InlineStoreBackend, objects are already converted to their respective config types.
-        if isinstance(value, DatasourceConfig):
+        if isinstance(value, (DatasourceConfig, FluentDatasource)):
             return value
         elif isinstance(value, dict):
+            # presence of a 'type' field means it's a fluent datasource
+            type_ = value.get("type")
+            if type_:
+                datasource_model = _SourceFactories.type_lookup.get(type_)
+                assert datasource_model, f"Unknown Datasource 'type': '{type_}'"
```

Comment:
I think raising may be preferable to an assert here.

Reply:
I'll raise `LookupError` or a `TypeError` here